internal/provider/resources/variable.go

```diff
@@ -120,6 +125,8 @@ func (r *VariableResource) Schema(_ context.Context, _ resource.SchemaRequest, r
 				Description: "Tags associated with the variable",
 				ElementType: types.StringType,
 				Optional:    true,
+				Computed:    true,
+				Default:     listdefault.StaticValue(defaultEmptyTagList),
```

Comment:
since this is a list type, there are these special constructs you need to use to set default values here

https://developer.hashicorp.com/terraform/plugin/framework/handling-data/attributes/list#plan-modification